modules/lazylibrarian.py

```diff
@@ -0,0 +1,358 @@
+    def AddAuthor(self, id, searchtype, **kwargs):
+
+        if searchtype == "authorId":
+            return self.fetch('addAuthor&name=%s' % id.replace (" ", "+"), text=True)
+        else:
+            return self.fetch('queueBook&id=%s' % id, text=True)
+
+    @cherrypy.expose()
+    @cherrypy.tools.json_out()
+    @require()
+    def GetHistoryList(self):
+        return self.fetch('getHistory')
+
+    @cherrypy.expose()
+    @require()
+    def GetBookArt(self, id):
+        return self.fetch('getBookArt&id=%s' % id, img=True)
+
+    @cherrypy.expose()
+    @require()
+    def GetBook(self, id):
+        return self.fetch('getBook&id=%s' % id)
+
+    @cherrypy.expose()
+    @require(member_of(htpc.role_user))
+    def ForceSearch(self):
+        return self.fetch('forceSearch', text=True)
+
+    @cherrypy.expose()
+    @require(member_of(htpc.role_user))
+    def ForceProcess(self, dir=None):
+        if dir:
+            return self.fetch('forceProcess?dir=%s' % dir, text=True)
+        return self.fetch('forceProcess', text=True)
+
+    @cherrypy.expose()
+    @require(member_of(htpc.role_user))
+    def ForceActiveAuthorsUpdate(self):
+        return self.fetch('forceActiveAuthorsUpdate', text=True)
+
+    @cherrypy.expose()
+    @require(member_of(htpc.role_user))
+    def ShutDown(self):
+        return self.fetch('shutdown', text=True)
+
+    @cherrypy.expose()
+    @require(member_of(htpc.role_user))
+    def UpDate(self):
+        return self.fetch('update', text=True)
+
+    @cherrypy.expose()
+    @require(member_of(htpc.role_user))
+    def ReStart(self):
+        return self.fetch('restart', text=True)
+
+    @cherrypy.expose()
+    @cherrypy.tools.json_out()
+    @require()
+    def Choose_Specific_Download(self, id):
+        return self.fetch('choose_specific_download&id=%s' % id)
+
+    @cherrypy.expose()
+    @cherrypy.tools.json_out()
+    @require()
+    def Download_Specific_Release(self, id, title, size, url, provider, kind):
+        return self.fetch('download_specific_release&id=%s&title=%s&size=%s&url=%s&provider=%s&kind=%s' %(id, title, size, url, provider, kind))
+
+    def fetch(self, command, url=None, api_key=None, img=False, json=True, text=False):
+        url = Lazylibrarian._build_api_url(command, url, api_key)
+        try:
+            # So shitty api..
+            if img or text:
+                json = False
+            result = ''
+            self.logger.debug('calling api @ %s' % url)
+            response = requests.get(url, timeout=30, verify=False)
+
+            if response.status_code != 200:
+                response.raise_for_status()
+                self.logger.error('failed to contact lazylibrarian')
+                return
+
+            if text:
+                result = response.text
+
+            if img:
+                result = response.content
+
+            if json:
+                result = response.json()
+
+            self.logger.debug('Response: %s' % result)
```

Comment:
Hi There, This is required for decoding the response? I'm pretty sure we use text, json and img responses, I dont know python too well but I think we need to keep this?